internal/handler/handler.go

```diff
@@ -135,8 +137,32 @@ func (h *Handler) Run(ctx context.Context) {
 
 // Handle processes a request.
 func (h *Handler) Handle(ctx context.Context, req *middleware.Request) (*middleware.Response, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/5046
-	panic("not implemented")
+	switch {
+	case req.OpMsg != nil:
+		doc, err := req.OpMsg.Document()
+		if err != nil {
+			return nil, err
+		}
+
+		msgCmd := doc.Command()
+
+		var cmdHandler middleware.HandleFunc
+
+		cmd, ok := h.commands[msgCmd]
+		if !ok {
+			cmdHandler = notFound(msgCmd)
+		} else if cmd.Handler == nil {
```

Comment:
We should be able to rewrite it with `else if` by using `switch`, early `return`, or something. `} else {` is generally not good, but `else if` is the worst.

But also, I think that branch is not possible: `initCommands` already set all handlers and uses `notImplemented` if needed.

Reply:
Thanks you are right, it's not possible to reach `notImplemented` here.